src/detect-ssl-version.c

```diff
@@ -218,13 +215,12 @@ static DetectSslVersionData *DetectSslVersionParse(DetectEngineCtx *de_ctx, cons
         SCLogError("Invalid empty value");
         goto error;
     }
+    if (tmp_str[0] == '!') {
+        ssl->negate = true;
+        tmp_str++;
+    }
     // iterate every version separated by comma
     while (tmp_str[0] != 0) {
-        uint8_t neg = 0;
-        if (tmp_str[0] == '!') {
-            neg = 1;
-            tmp_str++;
-        }
```

Comment:
~~will this work even if the second mentioned version is negated? e.g. (modified from the docs above) `ssl_version:sslv2,!sslv3;`~~